### DISCUSSION TITLE
Hardware Watchdog Reset on Scripting Ir Receiver Example

### DISCUSSION THREAD

**Discussion by odibox** [arendst/Tasmota#17194] [Q&A]
Hello
I have an ESP8266 NodeMcu V3 Board with self compiled Tasmota 12.1.1(TasmoCompiler-esp8266generic with IR and Scripting).

Now I have connected an IR Receiver to D7 GPIO13 and the Receiver works good.

But when I load the example Script from your website, then the board resets because of an hardware watchdog.
https://tasmota.github.io/docs/Scripting-Language/#multiple-ir-receiver-synchronization

Here is the example.
Now when I press the IR "ON" Button. The board resets because of an hardware watchdog.

```
>E
; get ir data
istr=IrReceived#Data

; on
if istr=="0x00F7C03F"
then
->WebSend [10.10.10.77]  power1 1
endif
istr=""
```

When I change the WebSend Command to a print %istr% then I can see the Ir Command in the console and the board works stable.

I have an other board ESP12E with an RGBCCT Controller and an IR Receiver on it. With the same self compiled Tasmota 12.1.1

```
>E
; get ir data
istr=IrReceived#Data

; on
if istr=="0x00F7C03F"
then
->power1 1
endif
istr=""
```

Now when I press the IR "ON" Button. The board resets because of an hardware watchdog.

Could you give me a hint, what caused this hardware watchdog.

Thanks in advice.

Andreas

**Comment by barbudor** [comment]
No response from  Tasmota at 10.10.10.77
Does the web send command works in the console?

**Reply by odibox** [reply]
From the console, Websend works. 
When the command (->WebSend [10.10.10.77]  power1 1) is in Section F or S, it also works.
But not when it's in section E, like in the example. Then the board resets because of an hardware watchdog.

When I use Global Vars instead of Websend, then it also works.

**Comment by odibox** [comment]
On self compiled Version Tasmota 9.5.0 it works like it should. (>E   ->WebSend [10.10.10.77] power1)

On 12.2.0, 11.1.0 and 10.2.0. it reboots.

00:00:00.001 HDW: ESP8266EX
00:00:00.063 Project tasmota - Tasmota Version 12.2.0(ir)-2_7_4_9(2022-11-28T12:28:32)
12:42:03.044 RSL: INFO3 = {"Info3":{"RestartReason":{"Exception":2,"Reason":"Exception","EPC":["3fff6f5c","00000000","00000000"],"EXCVADDR":"3fff6f5c","DEPC":"00000000"},"BootCount":9}}

00:00:00.001 HDW: ESP8266EX
00:00:00.061 Projekt tasmota - Tasmota Version 11.1.0(ir)-2_7_4_9(2022.11.28 12:47:27)
12:50:40.700 RSL: INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":4}}

00:00:00.002 HDW: ESP8266EX
00:00:00.065 Projekt tasmota - Tasmota Version 10.1.0(ir)-2_7_4_9(2022.11.28 12:53:39)
12:56:34.042 RSL: INFO3 = {"Info3":{"RestartReason":"Hardware Watchdog"}}

**Reply by gemu2015** [reply]
i can confirm this issue. it may be related to low stack memory.
stack usage may have been changed with versions.
stack in >E is much lower then in >S  only 1,6k versus 2,7 (simply print %stack%)
you may check stack size in the older working version to check if this has changed.
since websend needs more stack it may crash.

i am not sure if this can be fixed easily.
you should do the websend in >F. 
this will not affect functionality since you reset istr="" immediately it will only be executed once.

**Reply by odibox** [reply]
Hello
Here is the Code running.
```
>D 25
istr=""
ws="WebSend [10.10.10.77]"

; event section
>E
; get ir data
istr=IrReceived#Data

print Stacksize %stack%

if istr=="0xF740BF"
then
->%ws% Power1 On
endif

; off
if istr=="0xF7C03F"
then
->%ws% Power1 Off
endif

istr=""
```

In Firware 9.5 it looks lite this.
```
14:02:04.587 Script: nv=0, tv=2, vns=8, ram=86
14:02:10.647 RSL: RESULT = {"IrReceived":{"Protocol":"NEC","Bits":32,"Data":"0xF740BF","DataLSB":"0xEF02FD","Repeat":0}}
14:02:10.650 Stacksize 2608.00
14:02:10.775 RSL: RESULT = {"WebSend":"Done"}
14:02:12.102 RSL: RESULT = {"IrReceived":{"Protocol":"NEC","Bits":32,"Data":"0xF7C03F","DataLSB":"0xEF03FC","Repeat":0}}
14:02:12.106 Stacksize 2608.00
14:02:12.185 RSL: RESULT = {"WebSend":"Done"}
14:02:13.437 RSL: RESULT = {"IrReceived":{"Protocol":"NEC","Bits":32,"Data":"0xF740BF","DataLSB":"0xEF02FD","Repeat":0}}
14:02:13.441 Stacksize 2608.00
14:02:13.526 RSL: RESULT = {"WebSend":"Done"}
```

In Firware 12.2 it looks lite this.
```
13:56:21.302 script compressed to 161 bytes = 67 %
13:56:21.368 Script: nv=0, tv=2, vns=8, vmem=86, smem=6200
13:56:28.451 RSL: RESULT = {"IrReceived":{"Protocol":"NEC","Bits":32,"Data":"0xF740BF","DataLSB":"0xEF02FD","Repeat":0}}
13:56:28.455 Stacksize 2672.00
13:56:28.655 RSL: RESULT = {"WebQuery":{"POWER":"ON"}}
13:56:28.657 Stacksize 976.00

--------------- CUT HERE FOR EXCEPTION DECODER ---------------

Exception (2):
epc1=0x3fff768c epc2=0x00000000 epc3=0x00000000 excvaddr=0x3fff768c depc=0x00000000

>>>stack>>>

ctx: sys
sp: 3fff1abc end: 3fffffb0 offset: 0190
```


I think that the E Section has a Problem with something like Recursion. 
Have a look at the Stacksize 976.00 Entry. There should be an RSL: RESULT = {"IrReceived Message in front of it.

**Reply by gemu2015** [reply]
fixed
